### PR TITLE
Rails 5 upgrade

### PR DESCRIPTION
**Pull request by orangethunder**
This pull request branches off my other pull request here: https://github.com/rubysg/rubysg-reboot/pull/258

I updated all gems in the Gemfile. We should just do Rails 5.0 for now and leave Rails 5.1 as the next step.

I ran rails app:update to do the boilerplate changes, then went through to check that there weren't any changes that needed to be kept (notably puma.rb and application.rb had some, so I undid those changes).

Apart from that there was just a change required to one of the specs to use newer functional controller syntax.

I tested this a bit, seems to run fine. I did not test the Slack invite functionality.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,10 +1,10 @@
 source 'https://rubygems.org'
 
-ruby '2.4.0'
+ruby '2.5.1'
 
 gem 'puma'
-gem 'rails', '~> 4.2'
-gem 'pg', '0.21.0'
+gem 'rails', '~> 5.0.0'
+gem 'pg'
```
Comment (Drenmi) [inline Gemfile]: The `1.x` version of the PG gem does not work with Rails.

Comment (orangethunder) [reply Gemfile]: @Drenmi thanks, fixed now. Didn't pick it up since it seems the app doesn't actually use a DB? Is that planned for the future?

**Comment by hanmd82** [comment]
@orangethunder please merge upstream (with the ruby 2.5.1 upgrade) into this branch, so that the code diff reflects only the changes due to the upgrade to Rails 5.0

File: config/cable.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+development:
+  adapter: async
+
+test:
+  adapter: async
+
+production:
+  adapter: redis
+  url: redis://localhost:6379/1
```
Comment (hanmd82) [inline config/cable.yml]: this file can be commented out or removed, as this app is not using `action_cable`. I have spun up a a new Heroku test instance to verify that commenting it out does not cause any regressions

File: Gemfile (hunk 1 above)
Comment (hanmd82) [reply Gemfile]: Supposedly, Rails 5.1.5 onwards is compatible with `pg-1.0.0` https://github.com/rails/rails/issues/31673#issuecomment-372099518

This fix could come free with an upgrade to Rails 5.1

However we still need to evaluate https://github.com/rubysg/rubysg-reboot/issues/261

File: config/cable.yml (hunk 2 above)
Comment (hanmd82) [inline config/cable.yml]: @orangethunder could you remove this file?

**Comment by hanmd82** [comment]
please also pull `master` branch to sync changes

**Pull request merged**